contrib/src/main/java/com/datatorrent/contrib/parser/CsvParser.java

```diff
@@ -62,247 +71,193 @@
+    } catch (IOException e) {
+      logger.error("Exception in process method {}", e.getMessage());
+      DTThrowable.rethrow(e);
     }
-
-  }
-
-  /**
-   * Gets the array list of the fields, a field being a POJO containing the name
-   * of the field and type of field.
-   * 
-   * @return An array list of Fields.
-   */
-  public ArrayList<Field> getFields()
-  {
-    return fields;
-  }
-
-  /**
-   * Sets the array list of the fields, a field being a POJO containing the name
-   * of the field and type of field.
-   * 
-   * @param fields
-   *          An array list of Fields.
-   */
-  public void setFields(ArrayList<Field> fields)
-  {
-    this.fields = fields;
   }
 
-  /**
-   * Gets the delimiter which separates fields in incoming data.
-   * 
-   * @return fieldDelimiter
-   */
-  public int getFieldDelimiter()
+  @Override
+  public KeyValPair<String, String> processErorrTuple(byte[] input)
   {
-    return fieldDelimiter;
+    //This method is not invoked for CSV parser
+    return null;
   }
 
   /**
-   * Sets the delimiter which separates fields in incoming data.
-   * 
-   * @param fieldDelimiter
+   * Returns array of cellprocessors, one for each field
    */
-  public void setFieldDelimiter(int fieldDelimiter)
+  private CellProcessor[] getProcessor(List<Field> fields)
   {
-    this.fieldDelimiter = fieldDelimiter;
+    CellProcessor[] processor = new CellProcessor[fields.size()];
+    int fieldCount = 0;
+    for (Field field : fields) {
+      processor[fieldCount++] = CellProcessorBuilder.getCellProcessor(field.getType(), field.getConstraints());
+    }
+    return processor;
   }
 
-  /**
-   * Gets the delimiter which separates lines in incoming data.
-   * 
-   * @return lineDelimiter
-   */
-  public String getLineDelimiter()
+  @Override
+  public void teardown()
   {
-    return lineDelimiter;
+    try {
+      csvMapReader.close();
+    } catch (IOException e) {
+      logger.error("Error while closing csv map reader {}", e.getMessage());
+    }
+    try {
+      csvBeanReader.close();
+    } catch (IOException e) {
+      logger.error("Error while closing csv bean reader {}", e.getMessage());
+    }
   }
 
   /**
-   * Sets the delimiter which separates line in incoming data.
+   * Complete hdfs path of schema
    * 
-   * @param lineDelimiter
+   * @return
    */
-  public void setLineDelimiter(String lineDelimiter)
+  public String getSchemaPath()
   {
-    this.lineDelimiter = lineDelimiter;
+    return schemaPath;
   }
 
   /**
-   * Gets the name of the fields with type and format ( for date ) as comma
-   * separated string in same order as incoming data. e.g
-   * name:string,dept:string,eid:integer,dateOfJoining:date|dd/mm/yyyy
+   * Set path of schema
    * 
-   * @return fieldInfo
+   * @param schemaPath
+   *          path of the schema file in hdfs
    */
-  public String getFieldInfo()
+  public void setSchemaPath(String schemaPath)
```

Comment:
I it a requirement that this needs to be retrieved from HDFS. User might have a file in local file system.

I would suggest, reading config files from FileSystems in not responsibility of any operator. The file should be read by application and the content in string/byte[] format should be passed to the operator.